### DISCUSSION TITLE
Github action successful even though maven build failed

### DISCUSSION THREAD

**Discussion by mikenakis** [community/community#23482] [New to GitHub]
<p>I have this run of a workflow called “Release-Workflow”:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/mikenakis/Bathyscaphe/runs/6575192012">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/mikenakis/Bathyscaphe/runs/6575192012" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/12502657/181077339-259da111-8924-41aa-a354-845affc2f794.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://github.com/mikenakis/Bathyscaphe/runs/6575192012" target="_blank" rel="noopener nofollow ugc">Build software better, together</a></h3>

  <p>GitHub is where people build software. More than 83 million people use GitHub to discover, fork, and contribute to over 200 million projects.</p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
The workflow appears as having succeeded.<br>
The workflow contains one job, called “release”, which appears as having succeeded.<br>
The job contains several steps, one of which is called “Make-Release-Step”, which appears as having succeeded.<br>
But if you look inside, near the end, you will see that the maven build has miserably failed.<br>
I am not asking how to fix it so that maven does not fail; I already know what causes maven to fail and how to fix it.<br>
The problem is that if maven fails, then the step should fail, the job should fail, and the entire action should fail, no?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="mikenakis" data-post="1" data-topic="252440">
<div class="title">
<div class="quote-controls"></div>
 mikenakis:</div>
<blockquote>
<p>The problem is that if maven fails, then the step should fail, the job should fail, and the entire action should fail, no?</p>
</blockquote>
</aside>
<p>If a command exits with a non-zero exit status, yes. The thing is that you pipe the <code>mvn</code> output into <code>grep</code>, and as the <a href="https://www.gnu.org/software/bash/manual/html_node/Pipelines.html">Bash documentation on pipelines</a> says:</p>
<blockquote>
<p>The exit status of a pipeline is the exit status of the last command in the pipeline, unless the <code>pipefail</code> option is enabled (see <a href="https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html">The Set Builtin</a>). If <code>pipefail</code> is enabled, the pipeline’s return status is the value of the last (rightmost) command to exit with a non-zero status, or zero if all commands exit successfully.</p>
</blockquote>
<p>The <code>grep -v</code> didn’t find anything matching the expression you gave, so it exits with zero. If you want to keep the <code>grep</code>, your best option is probably to add <code>set -o pipefail</code> to the <code>run</code> step.</p>

**Comment by mikenakis** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a> thank you.</p>
